plugins/module_utils/cmd_runner.py

```diff
@@ -0,0 +1,290 @@
+# -*- coding: utf-8 -*-
+# (c) 2022, Alexei Znamensky <russoz@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+from functools import wraps
+
+from ansible.module_utils.common.collections import is_sequence
+from ansible.module_utils.six import iteritems
+
+
+def _ensure_list(value):
+    return list(value) if is_sequence(value) else [value]
+
+
+def _process_as_is(rc, out, err):
+    return rc, out, err
+
+
+class CmdRunnerException(Exception):
+    pass
+
+
+class MissingArgumentFormat(CmdRunnerException):
+    def __init__(self, arg, args_order, args_formats):
+        self.args_order = args_order
+        self.arg = arg
+        self.args_formats = args_formats
+
+    def __repr__(self):
+        return "<MissingArgumentFormat: {0} order: {1} formats: {2}>".format(
+            self.arg,
+            self.args_order,
+            self.args_formats,
+        )
```

Comment:
Shouldn't this rather be:
```suggestion
        return "MissingArgumentFormat({0!r}, {1!r}, {2!r})".format(
            self.arg,
            self.args_order,
            self.args_formats,
        )
```
so you can actually evaluate the `__repr__` output to get the object back?

(Same for the other `__repr__`s below.)

Reply:
That's making little sense to me. If we have an `except MissingArgumentFormat e:`, why would we want:

```python
    r = repr(e)
    obj = eval(r)
```

So that `obj` and `e` are the same thing? And particularly in the case of an exception, which is 99.99% of the times meant to be a very short-lived object, I see very little to gain from that.

Reply:
Ok, I did my reading, I see the recommendation for it, and will be changing. For the record, I still think the usefulness of that is debatable, but not going to pursue that argument.

Reply:
I also think it's not very useful, but if you choose to implement `__repr__` separately from `__str__`, you should probably do it correctly.